ingestion/src/metadata/ingestion/source/dashboard/tableau/metadata.py

```diff
@@ -364,58 +321,77 @@ def yield_dashboard(
             logger.debug(traceback.format_exc())
             logger.warning(f"Error to yield dashboard for {dashboard_details}: {exc}")
 
-    def yield_dashboard_lineage_details(
-        self, dashboard_details: TableauDashboard, db_service_name: str
+    def yield_dashboard_lineage(
+        self, dashboard_details: TableauDashboard
     ) -> Optional[Iterable[AddLineageRequest]]:
-        """
-        Get lineage between dashboard and data sources
-        """
 
-        data_source = next(
-            (
-                data_source
-                for data_source in self.workbook_datasources or []
-                if data_source.get("luid") == dashboard_details.id
-            ),
-            None,
-        )
-        to_fqn = fqn.build(
-            self.metadata,
-            entity_type=LineageDashboard,
-            service_name=self.config.serviceName,
-            dashboard_name=dashboard_details.id,
-        )
-        to_entity = self.metadata.get_by_name(
-            entity=LineageDashboard,
-            fqn=to_fqn,
-        )
+        yield from self.yield_datamodel_dashboard_lineage(dashboard_details) or []
 
-        try:
-            upstream_tables = data_source.get("upstreamTables")
-            for upstream_table in upstream_tables:
-                database_schema_table = fqn.split_table_name(upstream_table.get("name"))
-                from_fqn = fqn.build(
-                    self.metadata,
-                    entity_type=Table,
-                    service_name=db_service_name,
-                    schema_name=database_schema_table.get(
-                        "database_schema", upstream_table.get("schema")
-                    ),
-                    table_name=database_schema_table.get("table"),
-                    database_name=database_schema_table.get("database"),
-                )
-                from_entity = self.metadata.get_by_name(
-                    entity=Table,
-                    fqn=from_fqn,
+        for db_service_name in self.source_config.dbServiceNames or []:
+            yield from self.yield_dashboard_lineage_details(
+                dashboard_details, db_service_name
+            ) or []
+
+    def yield_datamodel_dashboard_lineage(
+        self, dashboard_details: TableauDashboard
+    ) -> Optional[Iterable[AddLineageRequest]]:
+        """
+        Args:
+            dashboard_details: Tableau Dashboard
+
+        Returns:
+            Lineage request between Data Models and Dashboards
+        """
+        om_dashboard: LineageDashboard = self._get_dashboard(dashboard_details)
```

Comment:
I might be missing something, but after every dashboard we should be able to pick it up from the context directly right? Not sure why we need to `self._get_dashboard`